### PR TITLE
cleanup ObjectHelper utility

### PR DESCRIPTION
**Pull request by bvahdat**

### CODE REVIEW

File: core/camel-util/src/main/java/org/apache/camel/util/ObjectHelper.java [hunk 1]
```diff
@@ -1151,7 +1138,8 @@ public static String lookupConstantFieldValue(Class<?> clazz, String name) {
      * @return       <tt>true</tt> if its a {@link Float#NaN} or {@link Double#NaN}.
      */
     public static boolean isNaN(Object value) {
-        return value instanceof Number && (FLOAT_NAN.equals(value) || DOUBLE_NAN.equals(value));
+        return value instanceof Number
```
Comment (davsclaus) [inline core/camel-util/src/main/java/org/apache/camel/util/ObjectHelper.java:1141]: I think you can check

if value instanceof Float
elseif ... Double

**Review by davsclaus — COMMENTED** [review]
There is also an ObjectHelper in camel-support

**Comment by bvahdat** [comment]
> There is also an ObjectHelper in camel-support

Thanks, yeah I've already noticed that copy as well, but that version doesn't seem to have the same logic as we have in `camel-util` module.

File: core/camel-util/src/main/java/org/apache/camel/util/ObjectHelper.java (hunk 1 above)
Comment (bvahdat) [reply core/camel-util/src/main/java/org/apache/camel/util/ObjectHelper.java]: Is this better now?

**Review by davsclaus — APPROVED** [review]

**Pull request merged**